### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
It will be nice if we provide the GitHub PR template.

Modifications:
- Add .github/PULL_REQUEST_TEMPLATE.md

Result:
- You can now easily write the PR description using this template.

### CODE REVIEW

File: site/src/pages/community/developer-guide.mdx [hunk 1]
```diff
@@ -386,3 +386,52 @@ try {
...
+- You can now monitor the state of all live threads and heap using `ManagementService`.
+
+In case of introducing a new feature, you can insert code snippet.
+```
+Server.builder()
+      .serviceUnder("/internal/management/", ManagementService.of());
+```
+
+Please keep in mind that we publish our [release notes](https://armeria.dev/release-notes/) using the
+result section of each PR. That is why writing good PR description is important.
```
Comment (ikhoon) [inline site/src/pages/community/developer-guide.mdx:435]: a?
```suggestion
result section of each PR. That is why writing a good PR description is important.
```

File: site/src/pages/community/developer-guide.mdx [hunk 2]
```diff
@@ -386,3 +386,52 @@ try {
...
+Describe the consequences that a user will face after this PR is merged.
+
+For example, if the PR fixes a bug, you can write: `You no longer see a Foo exception when using Bar.`
+If you introduce a new feature, you can write: `You can now do A using B.`
+These are examples:
+- You no longer see a `NullPointerException` when a request times out.
+- You can now monitor the state of all live threads and heap using `ManagementService`.
+
+In case of introducing a new feature, you can insert code snippet.
+```
```
Comment (ikhoon) [inline site/src/pages/community/developer-guide.mdx:429]:
```suggestion
```java
```

File: site/src/pages/community/developer-guide.mdx [hunk 3]
```diff
@@ -386,3 +386,52 @@ try {
     assertTrue(e.getMessage().contains("bad method"));
 }
 ```
+
+## How to write PR description
```
Comment (ikhoon) [inline site/src/pages/community/developer-guide.mdx:390]:
```suggestion
## How to write pull request description
```

File: site/src/pages/community/developer-guide.mdx [hunk 4]
```diff
@@ -386,3 +386,52 @@ try {
...
+
+There are three sections you need to write: `Motivation`, `Modifications`, and `Result`.
+
+### Motivation
+
+Explain why you're sending the PR and what's the problem you're trying to solve.
```
Comment (ikhoon) [inline site/src/pages/community/developer-guide.mdx:396]: From https://developers.google.com/tech-writing/one/words#use_acronyms_properly 😆
```suggestion
Explain why you're sending the pull request (PR) and what's the problem you're trying to solve.
```

**Review by ikhoon — APPROVED** [review]
Thanks! @minwoox

File: site/src/pages/community/developer-guide.mdx (hunk 4 above)
Comment (minwoox) [reply site/src/pages/community/developer-guide.mdx]: TIL 😄

File: site/src/pages/community/developer-guide.mdx [hunk 5]
```diff
@@ -386,3 +386,52 @@ try {
...
+## How to write pull request description
+
+There are three sections you need to write: `Motivation`, `Modifications`, and `Result`.
+
+### Motivation
+
+Explain why you're sending the pull request (PR) and what's the problem you're trying to solve.
+You do not have to include all the detail but please include as much as the reviewers can get the background
+of this change.
+If there's related GitHub issues, please link to them.
```
Comment (syleeeee) [inline site/src/pages/community/developer-guide.mdx:399]: Should be
-> If there are related GitHub issues ?

Comment (minwoox) [reply site/src/pages/community/developer-guide.mdx]: Thanks, fixed. 😉

File: site/src/pages/community/developer-guide.mdx [hunk 6]
```diff
@@ -386,3 +386,52 @@ try {
...
+In case of introducing a new feature, you can insert code snippet.
+```java
+Server.builder()
+      .serviceUnder("/internal/management/", ManagementService.of());
+```
+
+Please keep in mind that we publish our [release notes](https://armeria.dev/release-notes/) using the
+result section of each PR. That is why writing a good PR description is important.
+If this PR is something that we don't need to announce in the release notes such as cleaning up some logs,
+you can simply write it such as "Less verbose".
```
Comment (trustin) [inline site/src/pages/community/developer-guide.mdx:437]: Could be removed

Comment (minwoox) [reply site/src/pages/community/developer-guide.mdx]: Removed. 😉

File: site/src/pages/community/developer-guide.mdx [hunk 7]
```diff
@@ -386,3 +386,52 @@ try {
     assertTrue(e.getMessage().contains("bad method"));
 }
 ```
+
+## How to write pull request description
+
+There are three sections you need to write: `Motivation`, `Modifications`, and `Result`.
```
[the remark this answers is not in the document]
Comment (minwoox) [reply site/src/pages/community/developer-guide.mdx]: That's good writing. 👍

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+- Describe the modifications you've done.
+
+Result:
+
+- Closes #<GitHub issue number>. (If this resolves the issue.)
+- Describe the consequences that a user will face after this PR is merged.
+
+See [How to write pull request description](https://armeria.dev/community/developer-guide#how-to-write-pull-request-description)
+for more information.
```
[the remark this answers is not in the document]
Comment (minwoox) [reply .github/PULL_REQUEST_TEMPLATE.md]: Indeed, it's not a  markdown preview. Thanks!

**Review by trustin — APPROVED** [review]

**Pull request merged**